crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -722,6 +723,16 @@ impl<'w, 's, 'a> EntityCommands<'w, 's, 'a> {
         self
     }
 
+    /// Adds a [`Component`] to the entity.
+    pub fn insert_by_id(&mut self, id: ComponentId, component: impl Component) -> &mut Self {
+        self.commands.add(InsertById {
```

Comment:
```suggestion
        self.add(InsertById {
```
I think you can do that if `InsertById` implements `EntityCommand` instead?
I'm confused why `Insert` doesn't do that tbh, is it legacy code?
Maybe you can change that for `Insert` too, and for the other commands that have an `Entity` field?
If there are a lot of them don't bother.

Reply:
I've removed this `.commands` here and also in other commands that have an Entity field! (I'm not sure if I've removed everything)